### PR TITLE
Sales problem

### PR DESCRIPTION
**Pull request by Iamyuvraj143**

### CODE REVIEW

File: tax_helper.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+module TaxHelper
+
+  LIST_OF_EXEMPT_ITEMS = %w(food medicin chocolates pills book books chocolate )
+
+# to check is item exempt from tax
+  def exempt_good?(str)
+    item  = str.split(" ")
+    exempted_item = item & LIST_OF_EXEMPT_ITEMS
+   !exempted_item.empty?
```
Comment (kalpana19sharma) [inline tax_helper.rb:9]:
```suggestion
   exempted_item.present?
```
Should work?

File: tax_helper.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  # to check if the entered item is in valid forma
+  def is_valid_entry?(str)
+    quantity = str.split(" ")[0].to_i
+    price = str.split(" at ")[1].to_f
+    quantity == 0 || price == 0 ? false : true
```
Comment (kalpana19sharma) [inline tax_helper.rb]:
```suggestion
    !(quantity == 0 || price == 0)
```

File: tax_helper.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      break if input.include?("Done")
+      if is_valid_entry?(input) == false
+        puts "Invalid Entry 'Fatal'"
+        create_new_bill
+      end
+      items.append(input)
+    end  #end of while loop
+    items
+  end # end of create new bill
+end  # end module
```
Comment (kalpana19sharma) [inline tax_helper.rb:34]: Add new line

File: tax_problem.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+class Billing
+
+  def initialize(list_of_purchase)
+    @purchase_list = list_of_purchase
+  end
+
+  SALES_TAX = 5
+  BASIC_TAX = 10
+    
+  @purchase_list
```
Comment (kalpana19sharma) [inline tax_problem.rb]: What's the need of this?

File: tax_problem.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  def calculate_tax
+    for item in @purchase_list
```
Comment (kalpana19sharma) [inline tax_problem.rb]: Avoid using `for` loop

File: tax_problem.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+      if item_name.include?("imported")
+        sales_tax_amount = ((price * SALES_TAX) / 100 ).round(2)
+        total_tax_amount = total_tax_amount.to_f + sales_tax_amount.to_f
+      end
+
+      if exempt_good?(item) == false 
+        basic_tax_amount = ((price * BASIC_TAX) / 100 ).round(2)
+        total_tax_amount = total_tax_amount.to_f + basic_tax_amount.to_f
+      end
```
Comment (kalpana19sharma) [inline tax_problem.rb]:
```suggestion
tax_percentage = exempt_good?(item) ? 0 : BASIC_TAX
tax_percentage += SALES_TAX if item_name.include?("imported")

sales_tax_amount = ((price * tax_percentage) / 100 ).round(2)
total_tax_amount = total_tax_amount.to_f + sales_tax_amount.to_f
```

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: tax_helper.rb (hunk 1 above)
Comment (Iamyuvraj143) [reply tax_helper.rb:9]: ma'am I tried using present but it is not working !!

File: tax_problem.rb (hunk 4 above)
Comment (Iamyuvraj143) [reply tax_problem.rb]: @purchase_list removed

File: tax_problem.rb (hunk 5 above)
Comment (Iamyuvraj143) [reply tax_problem.rb]: Changed to " @purchase_list.each do |item|
                       end

File: tax_helper.rb (hunk 1 above)
Comment (kalpana19sharma) [reply tax_helper.rb:9]: ohh okay, `present?` is a Rails method. Ignore this comment.

File: tax_problem.rb (hunk 6 above)
Comment (Iamyuvraj143) [reply tax_problem.rb]: Acknowledged and changed !!

**Comment by Iamyuvraj143** [comment]
Feedback Changes Done

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**